ios/BurntModule.swift

```diff
@@ -107,118 +127,146 @@ struct ToastOptions: Record {
   
   @Field
   var message: String?
-
+  
   @Field
   var preset: ToastPreset = ToastPreset.done
-
+  
   @Field
   var duration: TimeInterval?
-
+  
   @Field
   var layout: ToastLayout?
-
+  
   @Field
   var shouldDismissByDrag: Bool = true
-
+  
   @Field
   var haptic: ToastHaptic = .none
-
+  
   @Field
   var from: ToastPresentSide = .top
+  
+  @Field
+  var icon: Icon? = nil
 }
 
-enum ToastHaptic: String, EnumArgument {
+enum ToastHaptic: String, Enumerable {
   case success
   case warning
   case error
   case none
-
+  
   func toSPIndicatorHaptic() -> SPIndicatorHaptic {
     switch self {
-    case .success:
-      return .success
-    case .warning:
-      return .warning
-    case .error:
-      return .error
-    case .none:
-      return .none
+      case .success:
+        return .success
+      case .warning:
+        return .warning
+      case .error:
+        return .error
+      case .none:
+        return .none
     }
   }
 }
 
-enum ToastPreset: String, EnumArgument {
+enum ToastPreset: String, Enumerable {
   case done
   case error
-
-  func toSPIndicatorPreset() -> SPIndicatorIconPreset {
+  case none
+  case custom
+  
+  func toSPIndicatorPreset(_ options: ToastOptions?) -> SPIndicatorIconPreset? {
     switch self {
-    case .done:
-      return .done
-    case .error:
-      return .error
+      case .done:
+        return .done
+      case .error:
+        return .error
+      case .custom:
+        return .custom(UIImage.init( systemName: options?.icon?.name ?? "swift")!.withTintColor(options?.icon?.color ?? .systemBlue, renderingMode: .alwaysOriginal))
+      case .none:
+        return .none
+        
     }
   }
 }
 
-enum ToastPresentSide: String, EnumArgument {
+enum ToastPresentSide: String, Enumerable {
   case top
   case bottom
-
+  
   func toSPIndicatorPresentSide() -> SPIndicatorPresentSide {
     switch self {
-    case .top:
-      return .top
-    case .bottom:
-      return .bottom
+      case .top:
+        return .top
+      case .bottom:
+        return .bottom
     }
   }
 }
 
 public class BurntModule: Module {
   public func definition() -> ModuleDefinition {
     Name("Burnt")
-
+    
     AsyncFunction("toastAsync") { (options: ToastOptions) -> Void in
-      let view = SPIndicatorView(title: options.title, message: options.message, preset: options.preset.toSPIndicatorPreset())
-
+      let view : SPIndicatorView
+      if(options.preset == .none){
+        view = SPIndicatorView(title: options.title, message: options.message)
+      } else if(options.preset == .custom){
+        view = SPIndicatorView(title: options.title, message: options.message, preset: options.preset.toSPIndicatorPreset(options)!)
```

Comment:
I believe it's better to unwrap if the values with `if let` the way I did previously, instead of asserting with `!`.

Reply:
I just noticed that it has only two ways to init it, so I just sorted out the code style, I think [this commit](https://github.com/nandorojo/burnt/pull/12/commits/8afee8e63352396df8c697da608a49bc96d07c71) will be good!  

![CleanShot 2023-03-03 at 3 24 59](https://user-images.githubusercontent.com/37520667/222657757-b6159624-89a0-4564-a73d-8c4858264e27.png)